### DISCUSSION TITLE
Instance styles don't override static styles

### DISCUSSION THREAD

**Discussion by Mehuge** [lit/lit#4650] [Q&A] [answered]
When using a combination of static styles and instance `<style>`s the cascading order seems back to front (to me). 

In the example below the text will be blue.  My brain tells me the text should be red. ie The instance style should override the static styles.

However, it doesn't.

```javascript
import {html, css, LitElement} from 'lit';

export class HelloWorld extends LitElement {
  static get styles() {
    return css`p { color: blue; }`;
  }

  render() {
    return html`
      <style>
         p { color: red; }
      </style>
      <p>This text should be red</p>
    `;
  }
}

customElements.define('hello-world', HelloWorld);
```
I was trying to do something like this:

```javascript
  render() {
    const styles = this.getStyleVariant();
    return html`
      ${styles && html`<style>${styles}</style>`}
      ${this.template()}
    `;
  }
```

But I guess that isn't going to work, as the variant will be overridden by the static styles.

The documentation doesn't seem to cover what happens when both static and instance styles are used.

**Comment by tbroyer** [comment, marked as answer]
`static styles` become _adopted stylesheets_ of the shadow root, and the spec (actually MDN here explaining the spec) [says](https://developer.mozilla.org/en-US/docs/Web/API/ShadowRoot/adoptedStyleSheets#:~:text=Stylesheets%20in%20the%20adoptedStyleSheets%20property%20are%20considered%20along%20with%20the%20shadow%20DOM%27s%20other%20stylesheets.%20For%20the%20purpose%20of%20determining%20the%20final%20computed%20CSS%20of%20any%20element%2C%20they%20are%20considered%20to%20have%20been%20added%20after%20the%20other%20stylesheets%20in%20the%20shadow%20DOM%20(ShadowRoot.styleSheets).):

> Stylesheets in the `adoptedStyleSheets` property are considered along with the shadow DOM's other stylesheets. For the purpose of determining the final computed CSS of any element, they are considered to have been added _after_ the other stylesheets in the shadow DOM ([`ShadowRoot.styleSheets`](https://developer.mozilla.org/en-US/docs/Web/API/ShadowRoot/styleSheets)).

Actual spec appears to be [here](https://drafts.csswg.org/cssom/#documentorshadowroot-final-css-style-sheets)

**Reply by Mehuge** [reply]
That confirms what I observed. 
Thanks.
